src/v/cloud_storage/remote.cc

```diff
@@ -485,6 +488,71 @@ ss::future<upload_result> remote::upload_controller_snapshot(
       std::nullopt);
 }
 
+ss::future<remote::jl_write_intent_upload_result>
+remote::jl_write_intent_upload(
+  const cloud_storage_clients::bucket_name& bucket,
+  uint64_t content_length,
+  const reset_input_stream& reset_str,
+  retry_chain_node& parent,
+  lazy_abort_source& lazy_abort_source,
+  std::optional<size_t> max_retries) {
+    uuid_t object_id{};
+    object_id.mutable_uuid() = boost::uuids::random_generator()();
+
+    // TODO(nv): https://en.wikipedia.org/wiki/Year_2038_problem
+    // Divide by 2**6 to get 64 second epochs. The maximum epoch
+    // that can be represented is 2**32 * 2**6 = 2**38 seconds which overflows
+    // around year 10686.
+    auto epoch = static_cast<uint64_t>(
```

Comment:
what is the epoch used for? if it's used for fencing, why are we not using a monotonic counter from raft?